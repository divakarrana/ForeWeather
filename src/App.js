import React, {useState, useEffect} from 'react';

import Header from './components/Header/header';
import Cards from './components/Cards/cards';


function App() {
  //Defining initial state to store weather data
  const [weatherData, setweatherData] = useState();

  //Fetching weather data from API and mapping to state

  //`http://api.openweathermap.org/data/2.5/forecast?id=1275339&units=metric&appid=${process.env.REACT_APP_WEATHER_API_KEY}`
  useEffect(() => {
    fetch("http://api.openweathermap.org/data/2.5/forecast?id=1275339&units=metric&appid=9036dc16bb6dfe2627295993204fcdca", {
      method: 'GET',
      headers: {
        'Content-Type': 'application/json',
        'Access-Control-Allow-Origin': '*'
      }})
      .then((response) => response.json())  
      .then((data) => setweatherData(filterDays(data.list)))
      .catch((error) => console.log(error.message));
  }, []);

  //Filtering day-wise data
  const filterDays = (weatherInfo) => {
    let prevDate = '';
    return weatherInfo.filter( weather => {
      let currDate = weather.dt_txt.split(" ")[0];
      if(currDate === prevDate){
        return false;
      }
      prevDate = currDate;
      return true;
    });
  }

  //Rendering root component and children
  return (
    <div className="App">
      <Header></Header>
      <Cards weatherData={weatherData}></Cards>
    </div>
     
  );
}

export default App;
