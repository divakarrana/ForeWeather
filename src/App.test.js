import React from 'react';
import ReactDOM from 'react-dom';
import {cleanup} from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import renderer from 'react-test-renderer';

import App from './App';

describe('<App />', () => {

  afterEach(cleanup);

  it("should render without crashing <App />", () => {
    const div = document.createElement('div');
    ReactDOM.render(<App></App>, div)
  });

  it("should match snapshot <App />", () => {
    const tree = renderer.create(<App></App>).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
