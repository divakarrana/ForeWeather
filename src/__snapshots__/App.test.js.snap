// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> should match snapshot <App /> 1`] = `
<div
  className="App"
>
  <div
    className="Header"
  >
    <div
      className="logoBox"
    >
      <i
        className="fas fa-smog"
      />
    </div>
    <h1
      className="appHeading"
    >
      Fore-Weather Â 
      <span
        className="subHeading"
      >
        5 day weather forecast for Mumbai
      </span>
    </h1>
  </div>
  <div
    className="container"
  >
    <div
      className="loading"
    >
      Gathering weather data... 
      <i
        className="fas fa-cog"
      />
    </div>
  </div>
</div>
`;
